tests/sentry/api/helpers/test_group_index.py

```diff
@@ -200,3 +202,15 @@ def test_ignore_with_substatus_until_escalating(self, send_robust):
         assert group.substatus == GroupSubStatus.UNTIL_ESCALATING
         assert send_robust.called
         assert not GroupInbox.objects.filter(group=group).exists()
+
+
+class TestHandleIsSubscribed(TestCase):
+    def test_is_subscribed(self):
```

Comment:
It might be worth also testing the update case, to show that the reason clears out.